app/src/main/java/com/example/spacex_demo_app/MainActivity.kt

```diff
@@ -1,39 +1,51 @@
 package com.example.spacex_demo_app
 
-import android.os.AsyncTask
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.os.Handler
 import android.os.Message
-import android.util.Log
 import android.widget.TextView
-import androidx.core.content.ContextCompat
+import androidx.appcompat.app.AppCompatActivity
 import com.apollographql.apollo.ApolloClient
-import com.apollographql.apollo.ApolloQueryCall
 import com.apollographql.apollo.api.Response
-import com.apollographql.apollo.exception.ApolloException
 import com.example.spacex_demo_app.api.Client
-import java.util.concurrent.Executor
-import java.util.concurrent.Executors
-import android.os.Handler as Handler
-import com.apollographql.apollo.ApolloCall as ApolloCall
+import com.example.spacex_demo_app.api.SpaceXApi
+import io.reactivex.rxjava3.annotations.NonNull
+import io.reactivex.rxjava3.core.Observer
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.observers.DisposableObserver
+
 
 class MainActivity : AppCompatActivity() {
 
+    private var disposable:Disposable? = null
+    private var disposable1:Disposable? = null
```

Comment:
use CompositeDisposable to dispose of all disposables with one method call on CompositeDisposable